lea/runner.py

```diff
@@ -131,6 +133,74 @@ def _expand_query(query):
             for selected in self.dag.select(q)
         }
 
+    def _split_query(self, query):
+        ast = sqlglot.parse_one(query, dialect=self.client.sqlglot_dialect)
+        ctes = {}
+        main_query = query
+
+        if isinstance(ast, sqlglot.exp.Select):
+            with_clause = ast.args.get("with")
+            if with_clause and isinstance(with_clause, sqlglot.exp.With):
+                for cte in with_clause.expressions:
+                    if isinstance(cte, sqlglot.exp.CTE):
+                        cte_name = cte.alias
+                        cte_query = cte.this.sql(dialect=self.client.sqlglot_dialect)
+                        ctes[cte_name] = cte_query
+
+                # Extract the main query (the part after the CTEs)
+                if "expression" in ast.args:
+                    main_query = ast.args["expression"].sql(dialect=self.client.sqlglot_dialect)
+                else:
+                    # If 'expression' is not in args, use the entire SELECT statement
+                    main_query = ast.sql(dialect=self.client.sqlglot_dialect)
+
+        self.log("Split query result:")
+        self.log(f"Number of CTEs found: {len(ctes)}")
+        for cte_name, cte_query in ctes.items():
+            self.log(f"CTE '{cte_name}': {cte_query[:100]}...")
+
+        return ctes, main_query
+
+    def _materialize_ctes_and_view(self, view):
+        self.log(f"Processing view: {view.key[0]}.{view.key[1]}")
+        self.log(f"Original query:\n{view.query}")
+
+        try:
+            ctes, main_query = self._split_query(view.query)
+        except Exception as e:
+            self.log(f"Error splitting query for {view.key[0]}.{view.key[1]}: {str(e)}")
+            self.log(f"Exception type: {type(e)}")
+            self.log(f"Exception args: {e.args}")
+            raise
+
+        for cte_name, cte_query in ctes.items():
+            try:
+                materialized_table = f"{view.key[0]}.{view.key[1]}__{cte_name}"
+                self.log(f"Attempting to materialize CTE: {materialized_table}")
+                self.log(f"CTE Query: {cte_query}")
+                cte_view = InMemorySQLView(
+                    key=tuple(materialized_table.split(".")), query=cte_query, client=self.client
+                )
+                self.client.materialize_view(cte_view)
+                self.log(f"Successfully materialized CTE: {materialized_table}")
+            except Exception as e:
+                self.log(f"Error materializing CTE {cte_name}: {str(e)}")
+                self.log(f"Exception type: {type(e)}")
+                self.log(f"Exception args: {e.args}")
+                raise
```

Comment:
Same, no need to raise an error here

Reply:
Thank you for the feedback, Max. I've removed the error handling at this level as suggested.